apps/src/code-studio/components/aiTutor/aiTutorChatMessagesTable.tsx

```diff
@@ -27,8 +63,122 @@ const AITutorChatMessagesTable: React.FunctionComponent<
     })();
   }, [sectionId]);
 
+  useEffect(() => {
+    const uniqueStudentFilterOptions = chatMessages.reduce<CheckboxOption[]>(
+      (acc, message) => {
+        const userId = `${message.userId}`;
+        if (!acc.some(student => student.value === userId)) {
+          acc.push({label: message.studentName, value: userId});
+        }
+        return acc;
+      },
+      []
+    );
+    setStudentFilterOptions(uniqueStudentFilterOptions);
+  }, [chatMessages]);
+
+  const handleStatusFilterChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const {value, checked} = event.target;
+    if (checked) {
+      setSelectedStatuses([...selectedStatuses, value]);
+    } else {
+      setSelectedStatuses(selectedStatuses.filter(status => status !== value));
+    }
+  };
+
+  const handleStudentFilterChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const {value, checked} = event.target;
+    if (checked) {
+      setSelectedUserIds([...selectedUserIds, value]);
+    } else {
+      setSelectedUserIds(selectedUserIds.filter(student => student !== value));
+    }
+  };
+
+  const filteredChatMessages = chatMessages.filter(message => {
+    const statusMatch =
+      selectedStatuses.length === 0 ||
+      selectedStatuses.includes(message.status);
+
+    const userMatch =
+      selectedUserIds.length === 0 ||
+      selectedUserIds.includes(`${message.userId}`);
+
+    const messageDate = moment(message.createdAt);
+    let timeMatch = false;
+    switch (selectedTimeFilter) {
+      case 'lastHour':
```

Comment:
is there a way for us to have these values as constants instead of hard-coding them in multiple places? Could we have an enum that we use to generate the formatted dropdown options?